### ISSUE TITLE
issue about echo -n option 

### ISSUE THREAD

**Issue by pollyduan** [yitsushi/totp-cli#95]
I encountered a strange issue where I used `echo GEZDGNBVMFRGGZDF | totp-cli instant` to generate a verification code. Other software was able to verify it successfully, but when I used `echo -n GEZDGNBVMFRGGZDF | totp-cli instant`, it failed. Based on my experience, `-n` should prevent the newline character from being displayed, so it should be correct. Why is it giving me an error instead?

```
$ echo -n GEZDGNBVMFRGGZDF | totp-cli instant
091943

$ echo GEZDGNBVMFRGGZDF | totp-cli instant
333229
```

The code generated by keepassxc is 333229

**Comment by pollyduan** [comment]
And the result of `totp-cli g myns myentry` is 333229 too. So I'm confused.

**Comment by yitsushi** [comment]
The code reads the token from `stdin` until newline. If there is no newline, it counts the content as empty string.

```
❯ cat check-95.sh
echo " --> With newline"
echo "GEZDGNBVMFRGGZDF" | go run . instant
echo " --> Without newline"
echo -n "GEZDGNBVMFRGGZDF" | go run . instant

❯ ./check-95.sh
 --> With newline
 -- token: "GEZDGNBVMFRGGZDF"
806679
 --> Without newline
 -- token: ""
230575
```

Technical details:
We use `bufio.NewReader` with `reader.ReadString('\n')` which "reads until the first occurrence of delim in the input". When it can't find a newline character before the stream ends we return with an error and and empty string.

Potential fix:
Return the result from `reader.ReadString('\n')` even if we meet an error.

**Comment by yitsushi** [comment]
@pollyduan : Now it's fixed in [v1.8.2](https://github.com/yitsushi/totp-cli/releases/tag/v1.8.2).

**Comment by pollyduan** [comment]
@yitsushi BINGO!! I checked, it's worked. THX a lot.
